src/v3/src/components/ConsentHeader/ConsentHeader.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Link } from '@mui/material';
+import {
+  Box, SettingsIcon, Typography,
+} from '@okta/odyssey-react-mui';
+import { escape } from 'lodash';
+import { Fragment, FunctionComponent, h } from 'preact';
+
+import { CONSENT_HEADER_STEPS, IDX_STEP } from '../../constants';
+import { useWidgetContext } from '../../contexts';
+import { useHtmlContentParser } from '../../hooks';
+import { getAppInfo, getHeadingReplacerFn, loc } from '../../util';
+
+const ConsentHeader: FunctionComponent = () => {
+  const { idxTransaction } = useWidgetContext();
+  const {
+    clientUri = undefined,
+    label = undefined,
+    logo = undefined,
+  } = idxTransaction ? getAppInfo(idxTransaction) : {};
+  const appName = escape(label);
+  const granularConsentTitle = loc(
+    'oie.consent.scopes.granular.title',
+    'login',
+    [appName],
+    {
+      $1: { element: 'h2', attributes: { class: 'no-translate' } },
+      $2: { element: 'p' },
+    },
+  );
+  const parsedGranularConsentTitle = useHtmlContentParser(
+    granularConsentTitle,
+    { replace: getHeadingReplacerFn({}, 'h2', 2, 3) },
+  );
+
+  if (!idxTransaction?.nextStep || !CONSENT_HEADER_STEPS.includes(idxTransaction.nextStep.name)) {
+    return null;
+  }
+  const stepName = idxTransaction.nextStep.name;
+
+  const getAppLogo = (altText: string, logoHref?: string) => (
+    typeof logoHref !== 'undefined'
+      ? (
+        <Box
+          component="img"
+          src={logoHref}
+          width="32px"
+          height="32px"
+          alt={altText}
+          className="client-logo custom-logo"
+          aria-hidden="true"
+        />
+      )
+      : (
+        // TODO: This is a temporary icon used until UX provides one
+        <SettingsIcon
+          titleAccess={altText}
+          classes="client-logo default-logo"
+          sx={{ width: '32px !important', height: '32px !important' }}
+          aria-hidden
+        />
+      )
+  );
+
+  const getAppIcon = () => {
+    const href: string | undefined = (clientUri as Record<string, string | undefined>)?.href;
```

Comment:
how come this type of `Record<string, string | undefined>` was needed?

Reply:
Ahh, that is due to the type of the `app` object values are `Record<string, unknown>` since the value of the record is unknown, we have to cast it to what it is expected to be, otherwise TS complains.

Reply:
Note that `clientUri` and `logo` can also be `undefined`. I don't think that will cause any runtime issues since we're using optional chaining but just be careful when casting since I think we're telling TS it won't be `undefined` when casting to this type.